src/installer/tests/Assets/TestProjects/HostApiInvokerApp/HostFXR.cs

```diff
@@ -142,6 +187,76 @@ static void Test_hostfxr_set_error_writer(string[] args)
             }
         }
 
+        /// <summary>
+        /// Test that invokes native api hostfxr_get_dotnet_environment_info.
+        /// </summary>
+        /// <param name="args[0]">hostfxr_get_dotnet_environment_info</param>
+        /// <param name="args[1]">(Optional) Path to the directory with dotnet.exe</param>
+        static void Test_hostfxr_get_dotnet_environment_info(string[] args)
+        {
+            string dotnetExeDir = null;
+            if (args.Length >= 2)
+                dotnetExeDir = args[1];
+
+            string hostfxr_version;
+            string hostfxr_commit_hash;
+            List<hostfxr.hostfxr_dotnet_environment_sdk_info> sdks = new List<hostfxr.hostfxr_dotnet_environment_sdk_info>();
+            List<hostfxr.hostfxr_dotnet_environment_framework_info> frameworks = new List<hostfxr.hostfxr_dotnet_environment_framework_info>();
+
+            int rc = hostfxr.hostfxr_get_dotnet_environment_info(
+                dotnet_root: dotnetExeDir,
+                reserved: IntPtr.Zero,
+                result: (info, result_context) => {
+                    hostfxr.hostfxr_dotnet_environment_info environment_info = Marshal.PtrToStructure<hostfxr.hostfxr_dotnet_environment_info>(info);
+
+                    hostfxr_version = environment_info.hostfxr_version;
+                    hostfxr_commit_hash = environment_info.hostfxr_commit_hash;
+                    for (int i = 0; i < environment_info.sdk_count; i++)
+                    {
+                        IntPtr pSdkInfo = new IntPtr(environment_info.sdks.ToInt64() + (i * Marshal.SizeOf<hostfxr.hostfxr_dotnet_environment_sdk_info>()));
+                        sdks.Add(Marshal.PtrToStructure<hostfxr.hostfxr_dotnet_environment_sdk_info>(pSdkInfo));
+                    }
+
+                    for (int i = 0; i < environment_info.framework_count; i++)
+                    {
+                        IntPtr pFrameworkInfo = new IntPtr(environment_info.frameworks.ToInt64() + (i * Marshal.SizeOf<hostfxr.hostfxr_dotnet_environment_framework_info>()));
+                        frameworks.Add(Marshal.PtrToStructure<hostfxr.hostfxr_dotnet_environment_framework_info>(pFrameworkInfo));
+                    }
+
+                    int env_info_size = Marshal.SizeOf(environment_info);
+                    if (env_info_size != environment_info.size)
+                        throw new Exception($"Size field value of hostfxr_dotnet_environment_info struct is {environment_info.size} but {env_info_size} was expected.");
+
+                    
+                    int env_sdk_info_size = (sdks.Count > 0) ? Marshal.SizeOf(sdks[0]) : -1;
+                    if (env_sdk_info_size != -1 && (env_sdk_info_size != sdks[0].size))
+                        throw new Exception($"Size field value of hostfxr_dotnet_environment_sdk_info struct is {sdks[0].size} but {env_sdk_info_size} was expected.");
```

Comment:
Nit: why not validate this when we add the SDK/framework to their respective lists? Each has this field, so it would make sense to validate all of them.